File: mysite/yelp/views/eventDetail.py
from django.shortcuts import render
from django.http import JsonResponse
import requests

def page(request, event_id, miles_away):
    API_KEY = 'NCmI5byYncKsu3ZKl5gze7IAS1Jzh6aF_JjLlTVBkLESa1UQCobyhrdastKevrkrOJr5ZBxYIQQLf_EWlPacIunyia0mw14VCgYbi9sjaOSPEa3AoDp9ti7CDmWZW3Yx'
    API_ENDPOINT = 'https://api.yelp.com/v3/events/' + event_id

    headers = {'Authorization': 'Bearer %s' %(API_KEY)}
    url_params = {}

    response = requests.request('GET', API_ENDPOINT, headers=headers, params=url_params)
    responseObj = response.json()

    d = {}
    d['name'] = responseObj['name']
    d['description'] = responseObj['description']
    d['time_start'] = responseObj['time_start']
    d['event_site_url'] = responseObj['event_site_url']
    d['latitude'] = responseObj['latitude']
    d['longitude'] = responseObj['longitude']

    if responseObj['cost'] is None:
        d['cost'] = 'Free!!!'
    else:
        d['cost'] = '$' + str(responseObj['cost'])

    return render(request, 'eventDetail.html',{'detail':d, 'miles_away':miles_away})

def api(request, event_id, miles_away):
    API_KEY = 'NCmI5byYncKsu3ZKl5gze7IAS1Jzh6aF_JjLlTVBkLESa1UQCobyhrdastKevrkrOJr5ZBxYIQQLf_EWlPacIunyia0mw14VCgYbi9sjaOSPEa3AoDp9ti7CDmWZW3Yx'
    API_ENDPOINT = 'https://api.yelp.com/v3/events/' + event_id

    headers = {'Authorization': 'Bearer %s' %(API_KEY)}
    url_params = {}

    response = requests.request('GET', API_ENDPOINT, headers=headers, params=url_params)
    responseObj = response.json()

    d = {}
    d['name'] = responseObj['name']
    d['description'] = responseObj['description']
    d['time_start'] = responseObj['time_start']
    d['event_site_url'] = responseObj['event_site_url']
    d['latitude'] = responseObj['latitude']
    d['longitude'] = responseObj['longitude']

    if responseObj['cost'] is None:
        d['cost'] = 'Free!!!'
    else:
        d['cost'] = '$' + str(responseObj['cost'])

    return JsonResponse({'detail':d, 'miles_away':miles_away})

File: react-yelp/src/components/eventDetail.js
import React, { Component } from 'react';
import {Button, Card, Grid, Loader, Dimmer} from 'semantic-ui-react';
import axios from 'axios';
import GoogleMapReact from 'google-map-react';

class eventDetail extends Component{
  state = {
    loadingData:false,
    detail:'',
    zoom:14,
  }

  async componentDidMount(){
    this.setState({loadingData:true});
    document.title = "EastVillage | Event Detail";

    var config = {};

    const url = 'https://eastvillage.pythonanywhere.com/json/event_detail/' + this.props.match.params.id + '&' + this.props.match.params.miles_away
    let response = await axios.get(url,config);
    this.setState({detail:response.data.detail});
    this.setState({loadingData:false});
  }

  render(){
    if(this.state.loadingData){
      return (
          <Dimmer active inverted>
            <Loader size='massive'>Loading...</Loader>
          </Dimmer>
      );
    }

    return (
      <Grid stackable>
        <Grid.Column width={10}>
          <Card fluid>
            <Card.Content>
              <Card.Header>{this.state.detail.name}</Card.Header>
              <Card.Meta style={{float:'right'}}>Time: {this.state.detail.time_start}</Card.Meta>
              <Card.Meta>Cost: {this.state.detail.cost}</Card.Meta>
              <p><b>Description:</b> {this.state.detail.description}</p>
              <Button style={{float:'right'}} primary basic href={this.state.detail.event_site_url} target='_blank'>Site Link</Button>
            </Card.Content>
          </Card>
        </Grid.Column>
        <Grid.Column width={6}>
          <div style={{height:'24em', width:'24em'}}>
            <GoogleMapReact
              bootstrapURLKeys={{ key:'AIzaSyC0dv2oxK00V51-FVp7UPrmK3kAxft6PEU'}}
              defaultCenter={{lat:this.state.detail.latitude, lng: this.state.detail.longitude}}
              defaultZoom={this.state.zoom}
            >
              
            </GoogleMapReact>
          </div>
        </Grid.Column>
      </Grid>
    );
  }
}

export default eventDetail;